### ISSUE TITLE
[a11y]: CheckList: Use proper heading levels

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/ibm-products#7171]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.62

### React version

18

### Automated testing tool and ruleset

accessibility-checker

### Description

If (for example), the application has a section with an `<h2>` heading, and that section contains one or more CheckList components, it's no good because the CheckList components will be using `<h2>` headings also, whereas in that case it should use `<h3>`. 

See https://www.w3.org/WAI/tutorials/page-structure/headings/ for details.

<img width="727" alt="Image" src="https://github.com/user-attachments/assets/d9ddbb2e-637a-4588-8617-644f2e13acc0" />

### WCAG 2.1 Violation

https://www.w3.org/TR/WAI-WEBCONTENT/wai-pageauth.html#tech-logical-headings

### Reproduction/example

https://stackblitz.com/edit/github-x5djhj-weiiynux?file=src%2Findex.scss

### Steps to reproduce

Just check heading levels using Chrome developer tools, both in https://stackblitz.com/edit/github-x5djhj-weiiynux?file=src%2Findex.scss and in Storybook.  In both cases they are a bit weird.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by anamikaanu96** [comment]
Closed via [7286](https://github.com/carbon-design-system/ibm-products/pull/7286)
